core/src/main/java/org/elasticsearch/index/seqno/SequenceNumbersService.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.index.seqno;
+
+import org.elasticsearch.index.IndexSettings;
+import org.elasticsearch.index.shard.AbstractIndexShardComponent;
+import org.elasticsearch.index.shard.ShardId;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+/** a very light weight implementation. will be replaced with proper machinery later */
+public class SequenceNumbersService extends AbstractIndexShardComponent {
```

Comment:
I really doubted for a long time if this should be an engine component or an indexShard level thing. I decided to go with IndexShard as I felt the engine is complicated enough and this can be done around it, I also expect it to have other integration points with the outside universe once we have the checkpointing logic (notification of failed seq# , return of local checkpoint and update of global checkpoints). Not sure if it's cleaner to point all of that at the Engine. I'll give it another thought and we can talk it through.